src/hooks/consortiumManager/useSettingMutation.js

```diff
@@ -0,0 +1,87 @@
+import { useMutation } from 'react-query';
+import { v4 as uuidv4 } from 'uuid';
+
+import { useOkapiKy } from '@folio/stripes/core';
+
+import { OKAPI_TENANT_HEADER } from '../../constants';
+import { throwErrorResponse } from '../../utils';
+import { usePublishCoordinator } from '../usePublishCoordinator';
+
+const injectTenantHeader = (ky, tenantId) => ky.extend({
+  hooks: {
+    beforeRequest: [
+      request => {
+        request.headers.set(OKAPI_TENANT_HEADER, tenantId);
+      },
+    ],
+  },
+});
+
+export const useSettingMutation = ({ path }) => {
+  const ky = useOkapiKy();
+  const { initPublicationRequest } = usePublishCoordinator();
+
+  const {
+    isLoading: isEntryCreating,
+    mutateAsync: createEntry,
+  } = useMutation({
+    mutationFn: ({ entry, tenants }) => {
+      const { shared, ...json } = entry;
+
+      if (shared) {
+        // TODO: implement creation of a shared setting
+        return Promise.reject(new Error('Not implemented yet'));
```

Comment:
`Promise.reject(new Error('Not implemented yet'))` can be moved to sepratate variable/function, used in 3 places

Reply:
it's todo :)